backend/src/main/java/de/neuefische/booklibrary/backend/api/ApiBookService.java

```diff
@@ -0,0 +1,98 @@
+package de.neuefische.booklibrary.backend.api;
+
+
+import de.neuefische.booklibrary.backend.Book;
+import de.neuefische.booklibrary.backend.BookState;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class ApiBookService {
+
+    private final String apiKey;
+    private final WebClient webClient;
+
+    public ApiBookService(@Value("${api.url}") String apiUrl, @Value("${api.key}") String apiKey) {
+        this.apiKey = apiKey;
+        this.webClient = WebClient.create(apiUrl);
+    }
+
+    String maxResults = "&maxResults=5";
```

Comment:
Consider defining the variable as final and write the name in UPPERCASE to signal it as a constant (If it actually does not change)